### PR TITLE
teleport.js should still function on avatars with no feet.

### LINKED ISSUE #10566: Blender Avatars cannot teleport
I am getting reports, and I can confirm that blender avatars are unable to teleport using the teleport.js . 

So avatars using the bone structure defined in the Blender Compendium Thread are not able to teleport using the controllers

### PR DESCRIPTION
**Pull request by hyperlogic**
Teleport.js would not work if avatar's were missing Leg or Feet joints.  Now we are more robust, we try to fallback to the Toe and Feet joints, and if those are missing we make a reasonable guess.

This should fix #10566 

QA Test plan: wear the following avatar:  http://west.nexusheavy.com/athauxan/PlumbotUpload/Plumbot_20.fst
Then attempt to teleport using your hand controllers.

### CODE REVIEW

**Comment by YuppyPlays** [comment]
Testing.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/395/HighFidelity-Beta-PR10613-7da73d0ff11832b66ed8a5e81e0e650391da4ae1.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/395/HighFidelity-Beta-PR10613-7da73d0ff11832b66ed8a5e81e0e650391da4ae1.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

with the avatar provided (Love the model by the way) all seemed to work perfectly :)

**Pass**

**Comment by nconn** [comment]
I have tested with this build, am able to teleport and all is well.
_And for those interested, this avatar will be submitted to the marketplace soon, feel free to do anything you like with it_

**Pass**

File: scripts/system/controllers/teleport.js [hunk 1]
```diff
@@ -366,45 +366,25 @@ function Teleporter() {
...
+        footJointIndex = MyAvatar.getJointIndex(FOOT_JOINT_NAMES[i]);
+        if (footJointIndex != -1) {
+            break;
+        }
+    }
+    if (footJointIndex != -1) {
+        // default vertical offset from foot to avatar root.
+        return -MyAvatar.getAbsoluteDefaultJointTranslationInObjectFrame(footJointIndex).y;
+    } else {
+        return DEFAULT_ROOT_TO_FOOT_OFFSET;
```
Comment (AndrewMeadows) [inline scripts/system/controllers/teleport.js]: Maybe this default value should be multiplied by `MyAvatar.scale`?

Comment (hyperlogic) [inline scripts/system/controllers/teleport.js]: Yeah, good catch, I'll fix that

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/441/HighFidelity-Beta-PR10613-90d241ca3c643ff5d0adec8afc5dbd4de83200b6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/441/HighFidelity-Beta-PR10613-90d241ca3c643ff5d0adec8afc5dbd4de83200b6.dmg

**Pull request merged**